Add Module report_element_visibility

hurrinico opened the pull request:
Adds new Module Report Element Visibility
Use Case:
I have an invoice with a huge number of lines, if i print the report i'll have a printing of many pages. In this case i may need to have on print's header the invoice's number in all the pages but first (because in first page invoice's number is already in). In this module i've created two new classes "not-first-page" and "not-last-page" to solve this issue.

Commit:
[ADD] Add Module report_qweb_element_page_visibility

eLBati commented:
:+1: thanks

pedrobaeza commented on report_element_visibility/README.rst:
```diff
@@ -0,0 +1,74 @@
+
+Install the module, then in the QWEB ``ir.ui.views`` used by your
+report, you can add an element with css class with not-first-page or not-last-page.
+For exemple if i need to improve invoice report header with
```
s/exemple/example

pedrobaeza replied:
s/i/you

pedrobaeza commented:
Module name should be _report_qweb_element_visibility_, because it only applies to QWeb.

pedrobaeza commented:
Can you add the reverse ones (first-page and last-page) to allow to have a report header and a report footer?

pedrobaeza commented on report_element_visibility/README.rst:
```diff
@@ -0,0 +1,74 @@
+=========================
+
+This module allows you to add 2 elements in QWEB reports:
+ - not-firt-page: Shows element in every page but first
```
firt/first

pedrobaeza commented on report_element_visibility/README.rst:
```diff
@@ -0,0 +1,74 @@
+
+To use this module, you need to:
+
+Install the module, then in the QWEB ``ir.ui.views`` used by your
```
Don't put "Install the module", because that's not something about the usage.

pedrobaeza replied:
I think this can be in Configuration section instead.

pedrobaeza commented on report_element_visibility/README.rst:
```diff
@@ -0,0 +1,74 @@
+report, you can add an element with css class with not-first-page or not-last-page.
+For exemple if i need to improve invoice report header with
+invoice's number in every page but first, and sale order report header 
+with order's name in every page but last, i'll add this code to external_layout_header::
```
s/i'll add/add

pedrobaeza commented on report_element_visibility/README.rst:
```diff
@@ -0,0 +1,74 @@
+with order's name in every page but last, i'll add this code to external_layout_header::
+
+    <t t-if="o._table=='account_invoice'">
+        <div class="not-first-page">
```
Why don't implement this like a method that you can evaluate with t-if?

hurrinico replied:
Because this is just an example of module's usage.
This module only adds classes "not-in-first-page" and "not-in-last-page".
But it's up to the user use them in header/footer or body's report.

pedrobaeza replied:
I'm talking of implementing classes as conditions, so you can make:

```
<t t-if="o._table=='account_invoice' and is_not_first_page()">
    <span t-esc="o.number"/>
</t>
```

hurrinico replied:
@pedrobaeza is_not_first_page() is a python or javascript method?
We have use this solution because with python we don't know how to get current page number, and because we think that a css class is more user's friendly.

pedrobaeza replied:
A python expression seems friendlier for me, but I have no idea if it can be achieved

tafaRU commented on report_element_visibility/__openerp__.py:
```diff
@@ -0,0 +1,37 @@
+#########################################################################
+
+{
+    'name': 'Report Extend Not in first Page',
```
Please change it accordingly its technical name.

pedrobaeza commented:
What about the module renaming and the other suggestion to add first-page/last-page? Thinking twice, even a better name is _report_qweb_element_page_visibility_

hurrinico commented:
@pedrobaeza  I've changed module's name and added first-page/last-page.
About "A python expression seems friendlier for me, but I have no idea if it can be achieved", this module is based on JavaScript because i need to know current page, first page and last page of a report, and i don't know how to achieve them by python

pedrobaeza commented on report_qweb_element_page_visibility/README.rst:
```diff
@@ -0,0 +1,76 @@
+Report Qweb ElementPage Visibility
+==================================
+
+This module allows you to add 4 elements in QWEB reports:
```
s/add 4 elements/use 4 classes

pedrobaeza commented on report_qweb_element_page_visibility/README.rst:
```diff
@@ -0,0 +1,76 @@
+==================================
+
+This module allows you to add 4 elements in QWEB reports:
+ - not-first-page: Shows element in every page but first
```
Put a line break above to allow the correct rendering and remove the initial space.

pedrobaeza commented on report_qweb_element_page_visibility/README.rst:
```diff
@@ -0,0 +1,76 @@
+To use this module, you need to:
+
+In the QWEB ``ir.ui.views`` used by your report,
+you can add an element with css class with not-first-page or not-last-page.
```
s/with.../with any of the classes described above

pedrobaeza commented:
Minors remarks, but :+1:

tafaRU commented on report_qweb_element_page_visibility/README.rst:
```diff
@@ -0,0 +1,76 @@
+ - not-first-page: Shows element in every page but first
+ - not-last-page: Shows element in every page but last
+ - first-page: Shows element only on first page
+ - last-page: Shows element only on last page
```
Please replace Show by show in all lines above.

hurrinico commented:
@pedrobaeza @tafaRU i've done the requested changes

pedrobaeza commented:
:+1:

tafaRU commented:
:+1:

gurneyalex commented:
nice addon! thanks :smile_cat:

leemannd commented:
Thank you
